torch/library.py

```diff
@@ -536,43 +553,95 @@ def wrap(f):
     return wrap
 
 
-@functools.singledispatch
-def impl(qualname, types, func=None, *, lib=None):
-    """Register an implementation for a device type for this operator.
+# @functools.singledispatch confuses mypy
```

Comment:
What's the exact bug, functools.singledispatch should be support by mypy now?

Reply:
The basic use is supported. The problem is that it's impossible to properly type this function without using overload - and it seems that using singledispatch + overload confuses mypy - it complains on the later use of @impl.register that the overloaded function has no attribute "register".

But looking at it I did realize I can make it simpler by only skipping the impl.register during TYPE_CHECKING.

Reply:
@aorenste I think it should work? You may just need to switch the order of singledispatch and overload I think?